xdsl/dialects/x86/ops.py

```diff
@@ -316,43 +346,42 @@ def custom_print_attributes(self, printer: Printer) -> Set[str]:
         return {"offset"}
 
 
-class R_M_Operation(
+class DM_Operation(
     Generic[R1InvT, R2InvT], X86Instruction, X86CustomFormatOperation, ABC
 ):
     """
-    A base class for x86 operations that have one register and one memory access with an optional offset.
+    A base class for x86 operations that load from memory (m, r2) into a destination
+    register (d, r1).
     """
 
-    r1 = operand_def(R1InvT)
+    r2 = operand_def(R2InvT)  # memory source
     offset = attr_def(IntegerAttr, default_value=IntegerAttr(0, 64))
-
-    result = result_def(R2InvT)
+    r1 = result_def(R1InvT)  # destination register
```

Comment:
note the `assembly_line_args` function below, the destination register is passed first, making it `r1`. This is the only major functional change in this file, the rest is just renaming.